.github/workflows/release-notes-generator.yml

```diff
@@ -16,24 +18,56 @@ on:
         type: string
         required: true
 
+env:
+  RELEASE_NOTES_FILE: "./release_notes.txt"
+
 jobs:
   get-release-notes:
     name: Get Release Notes for ${{ inputs.release_tag }} release
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout sui repo main branch
+      - name: Checkout ${{ inputs.previous_branch }}
```

Comment:
so instead of checkout and then `git rev-parse` you can directly call github api to get the commit sha,
for example:
`curl https://api.github.com/repos/MystenLabs/sui/commits/main | jq .sha`
and `curl https://api.github.com/repos/MystenLabs/sui/commits/releases/sui-v1.31.0-release | jq .sha`
no need for any token or anything because `sui` is a public repo.
Maybe calling API is faster but we don't need speed here either..so should be all good!

Reply:
Good point, simplified the logic